store/tooling/tooling.go

```diff
@@ -221,7 +231,61 @@ func (tsto *ToolingStore) DownloadSnap(name string, opts DownloadSnapOptions) (d
 	}
 	targetFn := filepath.Join(opts.TargetDir, baseName)
 
-	return tsto.snapDownload(targetFn, sar, opts)
+	if opts.OnlyComponents {
+		// No path, we use it to contain components information
+		downloadedSnap = &DownloadedSnap{
+			Path:            "",
+			Info:            sar.Info,
+			RedirectChannel: sar.RedirectChannel,
+		}
+	} else {
+		downloadedSnap, err = tsto.snapDownload(targetFn, sar, opts)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Download specified components
+	downloadedSnap.Components, err = tsto.downloadComponents(comps, sar, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	return downloadedSnap, err
+}
+
+func (tsto *ToolingStore) downloadComponents(comps []string, sar *store.SnapActionResult, opts DownloadSnapOptions) ([]*DownloadedComponent, error) {
+	downloadedComps := make([]*DownloadedComponent, len(comps))
+	for i, comp := range comps {
+		var srr *store.SnapResourceResult
+		for _, res := range sar.Resources {
+			if res.Name == comp {
+				srr = &res
+				break
+			}
+		}
+		if srr == nil {
+			return nil, fmt.Errorf("%s component for %s not found in store",
+				comp, sar.SnapName())
+		}
+		baseName := opts.Basename
+		if baseName == "" {
+			cpi := snap.MinimalComponentContainerPlaceInfo(
+				srr.Name, snap.R(srr.Revision), sar.SnapName())
+			baseName = cpi.Filename()
+		} else {
+			baseName += fmt.Sprintf("+%s.comp", srr.Name)
```

Comment:
this case doesn't seem to be tested but it seems the same logic is not tested for snaps either